karengir/Teamwork
# #169032077 add modification APIs

### what does this PR do?
Adds modification on APIs

### description of task to be completed?
 - Added jwt protection on the Api, added missing endpoints, added missing authentication
  - write tests for these functionalities

### How should this be manually tested?
- clone the repository https://github.com/karengir/Teamwork.git  and cd into the directory `/teamwork`
-checkout to branch ch-modofication-on-APIs-169032077
-Run command `npm run local` on your terminal
-using postman, 
-Navigate to `POST:/api/v1/auth/signup` to check if the password is returned
-Navigate to `POST:/api/v1/auth/signin` to sign In the account, using the creadentials from the account created above. In the response, the token is returned. copy that token
-Navigate to `POST:/api/v1/articles/` to create an article. first, without coping the token in the headers, you won't be able to create one. 
-copy the token in the headers in the Value column, and add Authorization in the key column, to create article
-Navigate to `GET: /api/v1/articles/1` to get an article with id:1
-Navigate to `GET:/api/v1/articles/feeds` to get all articles. 
-Navigate to `POST:/api/v1/auth/signup` to create another account; then to `POST:/api/v1/auth/signin` to log in that same account just created. 
-copy the token provide from the response
-Navigate to `DELETE:/api/v1/articles/1` "1" being the id of the article created by the first user.
-paste the token copied from the previous sign In in the headers, to delete the article
Navigate to`PATCH:DELETE:/api/v1/articles/2` "2" being the id of an article that does not exist yet, to edit the article.  

### What are the relevant pivotal tracker stories?
[#169032077](https://www.pivotaltracker.com/story/show/169032077)
![returnTokenSignIn](https://user-images.githubusercontent.com/53523700/66468209-12985080-ea86-11e9-952a-4de49df76c32.png)

Returns token on signIn : ![returnTokenOnSignIn](https://user-images.githubusercontent.com/53523700/66468045-cd741e80-ea85-11e9-98a8-12cb5d03adba.PNG)
No password returned in the response body: ![Capture](https://user-images.githubusercontent.com/53523700/66468046-ce0cb500-ea85-11e9-84e8-5a29c68f5331.PNG)
User should not be able to delete article they did not create: ![deleteArticleNotCreated](https://user-images.githubusercontent.com/53523700/66468047-ce0cb500-ea85-11e9-9a24-5dbfcae9891a.PNG)
When a user edits a non existing article: ![editArticleNotExisting](https://user-images.githubusercontent.com/53523700/66468048-cea54b80-ea85-11e9-9da5-fe077ef2cc18.PNG)
endpoint getting all the articles: ![getAllArticles](https://user-images.githubusercontent.com/53523700/66468049-cea54b80-ea85-11e9-8033-aa11d691d46a.PNG)
endpoint getting single article: and CreatedBy is defined: ![getSingleArticleEndpoint](https://user-images.githubusercontent.com/53523700/66468050-cf3de200-ea85-11e9-9678-9fe798910d89.PNG)
jwtProtection: ![jwtProtection](https://user-images.githubusercontent.com/53523700/66468052-cf3de200-ea85-11e9-868a-a5e868b04536.PNG)
![jwtProtectionWithToken](https://user-images.githubusercontent.com/53523700/66468053-cfd67880-ea85-11e9-9e35-09a9ce338420.PNG)
